aclk/agent_cloud_link.c

```diff
@@ -124,6 +124,10 @@ int cloud_to_agent_parse(JSON_ENTRY *e)
                     data->payload = mallocz(len+1);
                     if (!url_decode_r(data->payload, e->data.string, len + 1))
                         strcpy(data->payload, e->data.string);
+                    else {
```

Comment:
The `else` is the success case so this will not work.

Reply:
It copies the original string so that the final error will be generated by the webapi.

Reply:
So is there an error being sent back over the link in the response?

Reply:
There will the `code` set and the body will have the error message. It will send the same error code and text description that we get from typing the malformed url in `/api/v1/data?.....` 


Example

If the cloud requests a chart `system.lio` instead of `system.io` the following will be generated

(note this includes the TTL PR)


"\t{\"type\": \"http\",\n\t\"msg-id\": \"1fbde3c8-f9e4-4a0d-ad49-f0ae89e50884\",\n\t\"timestamp\": 1587396428,\n\t\"timestamp-offset-usec\": 346294,\n\t\"connect\": 1587396241,\n\t\"connect-offset-usec\": 135655,\n\t\"version\": 1,\n\t\"payload\": {\n\"code\": 404,\n\"body\": \"Chart is not found: system.lio\",\n\"headers\": \"HTTP/1.1 404 Not Found\\u000d\\nConnection: close\\u000d\\nServer: NetData Embedded HTTP Server v1.21.1-40-g37a963f3\\u000d\\nAccess-Control-Allow-Origin: *\\u000d\\nAccess-Control-Allow-Credentials: true\\u000d\\nContent-Type: text/plain; charset=utf-8\\u000d\\nDate: Thu, 01 Jan 1970 00:00:00 GMT\\u000d\\nCache-Control: no-cache, no-store, must-revalidate\\u000d\\nPragma: no-cache\\u000d\\nExpires: Thu, 01 Jan 1970 00:00:01 GMT\\u000d\\nContent-Length: 30\\u000d\\n\\u000d\\n\"\n}\n}"

Reply:
I didn't release that, then I will close the PR without merging it.